src/Generators/randomgenerators.jl

```diff
@@ -0,0 +1,407 @@
+"""
+struct ApproxErdosRenyi{T<:Integer, U<:Integer, R<:AbstractRNG} <: RandomGenerator
+    n::T
+    ne::U
+    rng::R
+end
+ApproxErdosRenyi(n, ne; rng=GLOBAL_RNG) = ApproxErdosRenyi(n, ne, rng)
+
+"""
+    struct ExpectedDegree <: RandomGenerator 
+
+A struct representing a generator for a random undirected graph in which two vertices `i` and `j` are connected
+with a probability (given by a vector of expected degrees `ω` indexed by vertex) of `ω[i]*ω[j]/sum(ω)`.
+
+### Required Fields
+- `ω::AbstractVector{<:Real}`: the probability vector for vertex connection
+
+### Optional Arguments
+- `rng::AbstractRNG`: set the random number generator (default: `Random.GLOBAL_RNG`)
+
+### Implementation Notes
+The algorithm should work well for `maximum(ω) ≪ sum(ω)`. As `maximum(ω)` approaches `sum(ω)`, some deviations
+from the expected values are likely.
+
+### References
+- Connected Components in Random Graphs with Given Expected Degree Sequences, Linyuan Lu and Fan Chung. [https://link.springer.com/article/10.1007%2FPL00012580](https://link.springer.com/article/10.1007%2FPL00012580)
+- Efficient Generation of Networks with Given Expected Degrees, Joel C. Miller and Aric Hagberg. [https://doi.org/10.1007/978-3-642-21286-4_10](https://doi.org/10.1007/978-3-642-21286-4_10)
+"""
+struct ExpectedDegree{T<:Real, U<:AbstractVector{T}, R<:AbstractRNG} <: RandomGenerator
+    ω::U
+    rng::R
+end
+ExpectedDegree(ω; rng=GLOBAL_RNG) = ExpectedDegree(ω, rng)
+
+"""
+    struct WattsStrogatz <: RandomGenerator
+
+A struct representing a generator for a [Watts-Strogatz](https://en.wikipedia.org/wiki/Watts_and_Strogatz_model)
+small model random graph with `n` vertices, each with degree `k`. Edges are
+randomized per the model based on probability `β`.
+
+### Required Fields
+- `n::Integer`: the number of vertices in the graph
+- `k::Integer`: the number of edges per vertex
+- `β::Real`: the probability of edge randomization (`0 < β ≤ 1`)
+
+### Optional Arguments
+- `rng::AbstractRNG`: set the random number generator (default: `Random.GLOBAL_RNG`)
+"""
+struct WattsStrogatz{T<:Integer, U<:Integer, V<:Real, R<:AbstractRNG} <: RandomGenerator
+    n::T
+    k::U
+    β::V
+    rng::R
+end
+WattsStrogatz(n, k, β; rng=GLOBAL_RNG) = WattsStrogatz(n, k, β, rng)
+
+"""
+    struct BarabasiAlbert <: RandomGenerator
+
+A struct representing a generator for a [Barabási–Albert model](https://en.wikipedia.org/wiki/Barab%C3%A1si%E2%80%93Albert_model)
+random graph with `n` vertices. It is grown by adding new vertices to an initial graph `g` with `k`
+vertices). Each new vertex is attached with `k` edges to `k` different vertices already present in
+the system by preferential attachment. Initial graphs are undirected and consist of isolated
+vertices by default.
+
+### Required Fields
+- `n::Integer`: the number of vertices in the final graph
+- `k::Integer`: the number of edges preferentially attached per vertex
+- `g::AbstractGraph`: an initial graph
+
```

Comment:
This is a breaking change. Before, we could pass in `n0` which was the number of vertices in a graph and `is_complete` to set it to a complete graph or false for empty. We could also pass in an existing graph. Now we just pass in a graph.